src/idiomatic/foundations-api-design/predictable-api/naming-conventions/02-push.md

```diff
@@ -0,0 +1,29 @@
+---
+minutes: 2
+---
+
+# Push
+
+Common on array-like structures.
+
+```rust
+use std::collections::VecDeque;
+
+let mut items: Vec<u8> = Vec::new();
+items.push(42);
+let mut items_queue: VecDeque<u8> = VecDeque::new();
+items_queue.push_front(233);
+// [233]
+items_queue.push_back(42);
+// [233, 42]
+items_queue.push_front(128);
+// [128, 233, 42]
+dbg!(items_queue);
```

Comment:
```suggestion
impl<T> Vec<T> {
  fn push(&mut self, value: T)
}

impl<T> VecDeque<T> {
  fn push_back(&mut self, value: T)
  fn push_front(&mut self, value: T)
}
```

I think three method signatures would communicate the idea a lot better than a usage example.